tests/media/media.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { media } from '@/media'
+import { isRef } from 'vue'
+
+describe('media', () => {
+    
+    beforeAll(() => {
+      Object.defineProperty(window, "matchMedia", {
+        writable: true,
+        value: jest.fn().mockImplementation(query => ({
+          matches: false,
+          media: query,
+          onchange: null,
+          addListener: jest.fn(), // Deprecated
+          removeListener: jest.fn(), // Deprecated
+          addEventListener: jest.fn(),
+          removeEventListener: jest.fn(),
+          dispatchEvent: jest.fn(),
+        }))
+      });
+    });
+
+    it('returns a ref', async () => {
+        const match = media('(hover)')
+
+        expect(isRef(match)).toBe(true)
+    })
+
+})
```

Comment:
does it make sense to test that on unmount it will remove event listener? I know you have other tests with vue components for testing unmount, just not sure how difficult it would be to check for existence of your event listener

Reply:
I was unable to figure out how to test this. I cannot spy on `unwatch` directly because its essentially private. Maybe you have some ideas here?